prow/config/config.go

```diff
@@ -78,6 +78,20 @@ type Config struct {
 
 	// PushGateway is a prometheus push gateway.
 	PushGateway PushGateway `json:"push_gateway,omitempty"`
+
+	// OwnersDirBlacklist is used to configure which directories to ignore when
+	// searching for OWNERS{,_ALIAS} files in a repo.
+	OwnersDirBlacklist OwnersDirBlacklist `json:"owners_dir_blacklist,omitempty"`
+}
+
+// OwnersDirBlacklist is used to configure which directories to ignore when
+// searching for OWNERS{,_ALIAS} files in a repo.
+type OwnersDirBlacklist struct {
+	// Repos configures a directory blacklist per repo (or org)
+	Repos map[string][]string `json:"repos"`
+	// Default configures a default blacklist for repos (or orgs) not
+	// specifically configured
+	Default []string `json:"default"`
```

Comment:
So right now the API is:

 - if a user provides no repos and no default, use the built-in default
 - if a user provides only a default, use the union of that and the built-in default
 - if a user provides a default and an org config, use the union of those with the built-in
 - if a user provides a default, an org config and a repo config, use the union of all those with the built-in

Do we have a concrete use-case for the user-provided default where we wouldn't just want to check that in as the built-in default?

/cc @cjwagner

Reply:
What I was thinking:
- If nothing at all is configured, behave like right now (-> built-in default)
- There was the discussion, if `vendor` should be excluded by default for all repos. This could be easily done then without a code change (-> `Default` in the config)
- Then we most probably want add additional excludes per repo / org (-> `Repos` in the config)